File: docs/sphinx/targets/python/qibo.py
import cudaq

# Set the target at the beginning of the program.
cudaq.set_target("qibo")


# Create the kernel.
@cudaq.kernel
def kernel():
    qvector = cudaq.qvector(2)
    h(qvector[0])
    x.ctrl(qvector[0], qvector[1])
    mz(qvector)


# Execute on synchronously on the Qibo cloud and print out the results.
counts = cudaq.sample(kernel)
print(counts)

File: docs/sphinx/targets/cpp/qibo.cpp
// Compile and run with:
// ```
// nvq++ --library-mode --target qibo qibo.cpp
// ./a.out
// ```

#include <cudaq.h>
#include <fstream>

struct bell_state {
  auto operator()() __qpu__ {
    cudaq::qvector q(2);
    h(q[0]);
    x<cudaq::ctrl>(q[0], q[1]);
    auto result = mz(q);
  }
};

int main() {
  // OR: Submit to Qibo synchronously.
  auto result_counts = cudaq::sample(bell_state{});
  result_counts.dump();
}
